cleangoldenami/main.go

```diff
@@ -115,26 +176,65 @@ func getPublicImages(ec2Client DescribeImagesClient) ([]ec2Types.Image, error) {
 	return describeImagesResponse.Images, nil
 }
 
-func getOldestImage(images []ec2Types.Image) (ec2Types.Image, error) {
+func hasVersionTag(image ec2Types.Image, tag string) bool {
+	for _, t := range image.Tags {
+		if *t.Key == "version" && *t.Value == tag {
+			return true
+		}
+	}
+	return false
+}
+
+func hasMatchingArchitecture(image ec2Types.Image, architecture ec2Types.ArchitectureValues) bool {
+	return image.Architecture == architecture
+}
+
+// filterImages filters AWS AMIs based on their version tag and architecture.
+func filterImages(images []ec2Types.Image) ([]ec2Types.Image, []ec2Types.Image, []ec2Types.Image) {
+	var arm64Images, legacyx86Images, x86Images []ec2Types.Image
+	for _, image := range images {
+
+		if hasVersionTag(image, "rhel-arm64") && hasMatchingArchitecture(image, "arm64") {
```

Comment:
the AWS SDK has constants we should use here for the architecture types. You can find them with `ec2Types.ArchitectureValuesX`